File: backend-java/minimarket-api-java/src/main/java/com/minimarket/api/controller/SupplierSyncController.java
package com.minimarket.api.controller;

import com.minimarket.api.dto.ApiMessageResponse;
import com.minimarket.api.dto.SupplierSyncRequestDto;
import com.minimarket.api.service.SupplierSyncService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/supplier-sync")
public class SupplierSyncController {

    private final SupplierSyncService supplierSyncService;

    public SupplierSyncController(SupplierSyncService supplierSyncService) {
        this.supplierSyncService = supplierSyncService;
    }

    // POST /api/supplier-sync -> ejecuta o previsualiza la sincronizacion del catalogo del proveedor.
    @PostMapping
    public ResponseEntity<?> sync(@RequestBody SupplierSyncRequestDto request) {
        var result = supplierSyncService.sync(request);
        if (!result.success()) {
            return ResponseEntity.badRequest().body(new ApiMessageResponse(result.error()));
        }

        return ResponseEntity.ok(result.data());
    }
}

File: backend/Minimarket.Api/Controllers/SupplierSyncController.cs
using Microsoft.AspNetCore.Mvc;
using Minimarket.Api.DTOs;
using Minimarket.Api.Services;

namespace Minimarket.Api.Controllers;

[ApiController]
[Route("api/supplier-sync")]
public class SupplierSyncController(ISupplierSyncService supplierSyncService) : ControllerBase
{
    // POST api/supplier-sync  -> ejecuta o previsualiza la sincronizacion del catalogo del proveedor.
    [HttpPost]
    public async Task<ActionResult<SupplierSyncResultDto>> Sync([FromBody] SupplierSyncRequestDto request)
    {
        var result = await supplierSyncService.SyncAsync(request);
        if (!result.Success)
        {
            return BadRequest(new { message = result.Error });
        }

        return Ok(result.Result);
    }
}
